glib-macros/src/variant_derive.rs

```diff
@@ -62,7 +63,7 @@ pub fn derive_variant_for_struct(
             let to_variant = quote! {
                 impl #generics #glib::ToVariant for #ident #generics {
                     fn to_variant(&self) -> #glib::Variant {
-                        #glib::Variant::tuple_from_iter(::std::array::IntoIter::new([
+                        #glib::Variant::tuple_from_iter(::std::array::IntoIter::<#glib::Variant, #idents_len>::new([
```

Comment:
This example highlights a notable point; because the function uses `x: impl Y` syntax rather than real generics, it's not as easy as using `tuple_from_iter::<Variant, _>` to help inference ([this is a known limitation of the syntax](https://doc.rust-lang.org/reference/types/impl-trait.html#anonymous-type-parameters))